tests/integration/test_gomod_packages.py

```diff
@@ -0,0 +1,31 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import utils
+
+
+def test_gomod_vendor_without_flag(test_env):
+    """
+    Validate failing of gomod vendor request without flag.
+
+    Checks:
+    * The request failed with expected error message
+    """
+    env_data = utils.load_test_data("gomod_packages.yaml")["vendored_without_flag"]
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"], test_env.get("timeout"))
+    initial_response = client.create_new_request(
+        payload={
+            "repo": env_data["repo"],
+            "ref": env_data["ref"],
+            "pkg_managers": env_data["pkg_managers"],
+        },
+    )
+    completed_response = client.wait_for_complete_request(initial_response)
+    if test_env["strict_mode_enabled"]:
```

Comment:
can we use `test_env.get("strict_mode_enabled")` instead? Then users would not need to change the test envi file in their testing setups (e.g., Jenkins) until they enable strict mode in production.